### PR TITLE
fix: Added save_load to squarcb

### PR DESCRIPTION
**Pull request by bassmang**

### CODE REVIEW

**Comment by ataymano** [comment]
Can you please add test here?
https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/test/RunTests

File: vowpalwabbit/cb_explore_adf_squarecb.cc [hunk 1]
```diff
@@ -280,6 +289,17 @@ void cb_explore_adf_squarecb::predict_or_learn_impl(VW::LEARNER::multi_learner&
   }
 }
 
+void cb_explore_adf_squarecb::save_load(io_buf& io, bool read, bool text)
+{
+  if (io.num_files() == 0) { return; }
+  if (!read || _model_file_version >= VERSION_FILE_WITH_CB_TO_CBADF)
```
Comment (jackgerrits) [inline vowpalwabbit/cb_explore_adf_squarecb.cc:295]: Please add a new constant that represents square cb serializing and deserializing the counter. It's fine for it to use the same verison but should have its own define.

File: vowpalwabbit/cb_explore_adf_squarecb.cc [hunk 2]
```diff
@@ -73,14 +80,16 @@ struct cb_explore_adf_squarecb
...
+    float gamma_scale, float gamma_exponent, bool elim, float c0, float min_cb_cost, float max_cb_cost,
+    VW::version_struct model_file_version)
     : _counter(0)
     , _gamma_scale(gamma_scale)
     , _gamma_exponent(gamma_exponent)
     , _elim(elim)
     , _c0(c0)
     , _min_cb_cost(min_cb_cost)
     , _max_cb_cost(max_cb_cost)
+    , _model_file_version(model_file_version)
```
Comment (jackgerrits) [inline vowpalwabbit/cb_explore_adf_squarecb.cc:92]: Avoid the extra copy
```suggestion
    , _model_file_version(std::move(model_file_version))
```

File: vowpalwabbit/vw_versions.h [hunk 3]
```diff
@@ -20,3 +20,4 @@
 #define VERSION_FILE_WITH_CCB_MULTI_SLOTS_SEEN_FLAG \
   "8.9.0"  // CCB optimization for models with only single slot used introduced in this version.
 #define VERSION_FILE_WITH_CB_TO_CBADF "8.11.0"  // first version that maps --cb to use cb_adf
+#define VERSION_FILE_WITH_SAVE_RESUME "8.11.0"  // version that ensures accuracy of loaded models in squarecb
```
Comment (jackgerrits) [inline vowpalwabbit/vw_versions.h:23]: This needs context to know what means. How about?
```suggestion
#define VERSION_FILE_WITH_SQUARE_CB_SAVE_RESUME "8.11.0"  // version that ensures accuracy of loaded models in squarecb
```

**Review by jackgerrits — APPROVED** [review]

**Review by olgavrou — APPROVED** [review]

**Pull request merged**